docs/content/Examples-Tutorials-Recipes/Enforcing-mandatory-filters.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Enforcing Mandatory Filters in a Query
+permalink: /recipes/enforcing-mandatory-filters
+category: Examples & Tutorials
+subCategory: queries
+menuOrder: 1
+---
+
+## Use case
+
+Let's imagine that on New Year's Eve, December 30th, 2019, we renamed our store,
+changed the design, and started selling completely different products. At the
+same time, we preserve the database from the previous store. Our goal is to only show
+orders and users created after December 30th, 2019. In the recipe below,
+we'll learn how to add mandatory filters to all queries.
+
+## Configuration
+
+To enforce mandatory filters we'll use the
+[`queryRewrite`](https://cube.dev/docs/security/context#using-query-rewrite)
+extension.
+
+To solve this, first we need to get all measures and
+dimensions from a query. Next, we add filters for the
+measures and dimensions. This will make sure we only show users and orders created after December 30th, 2019.
+
+```javascript
+module.exports = {
+  queryRewrite: (query) => {
```

Comment:
Are you getting the cube names here? I think code should be changed, at least variable names should be renamed, to be more explicit. I had hard time reading the code first.  Maybe adding comments. 

E.g. cubeNames from measures + cubeNames from dimensions. Also, I'm not sure it is correct. Consider the case where we have Products joined to orders in a single query, like Orders count by Product name, and we don't want to filter based on Products.createdAt or even Product don't have createdAt.

Reply:
>Are you getting the cube names here?  

Yes. I agree, it looks confusing... I'll change this code.

>Consider the case where we have Products joined to orders in a single query, like Orders count by Product name, and we don't want to filter based on Products.createdAt or even Product don't have createdAt.

@keydunov it's a good question. I thought about it, and I tried to find a use case relevant to our `ecom` database. Perhaps you have another use case for mandatory filters? 😊

Reply:
I think it's okay to simplify this recipe to filter on a single cube only (say, `Orders`). So, you'll be able to get rid of all this not-so-relevant cube-name-parsing code.

Reply:
However, I think it would be great to still have two different queries in this recipe that show that whatever members and dimensions are present, a filter will still be applied.

Reply:
@igorlukanin @keydunov I changed the use case and now I get orders in different status